Bump Recipe Versions

Bastian-Krause opened the pull request:
- python3-janus: version bump 0.6.1 -> 1.0.0
- python3-powerrelay: version bump bf5d2e1 -> 41c030f
- python3-usbmuxctl: install from pypi, version bump 096905f -> 0.1.3
- sispmctl: version bump 4.9 -> 4.11

Build-tested only.

Bastian-Krause committed:
python3-janus: version bump 0.6.1 -> 1.0.0

janus declared the PEP-517 compliant
`setuptools.build_meta build-backend` in the `[build-system]` section of
pyproject.toml a while ago [1].

Since that's the preffered way of building and installing Python
packages nowadays, let's use that corresponding
"python_setuptools_build_meta class" instead of "setuptools3".

[1] https://github.com/aio-libs/janus/commit/67e3782dc540ee13858027c3521691e779fcdc16

Signed-off-by: Bastian Krause <bst@pengutronix.de>

Bastian-Krause committed:
python3-powerrelay: version bump bf5d2e1 -> 41c030f

Signed-off-by: Bastian Krause <bst@pengutronix.de>

Bastian-Krause committed:
python3-usbmuxctl: install from pypi, version bump 096905f -> 0.1.3

Signed-off-by: Bastian Krause <bst@pengutronix.de>

Bastian-Krause committed:
sispmctl: version bump 4.9 -> 4.11

Signed-off-by: Bastian Krause <bst@pengutronix.de>

jluebbe reviewed (dismissed):
Looks good, but it should be run-time tested as well.

Bastian-Krause commented:
Agreed, maybe @hnez can do that.

hnez commented:
The files installed by `python3-pyserial-labgrid.bb` conflict with those installed by `python3-pyserial.bb` from e.g. [`meta-openembedded`](/openembedded/meta-openembedded/blob/langdale/meta-python/recipes-devtools/python/python3-pyserial_3.5.bb).
I couldn't quite figure out which of the packages in [`meta-lxatac`](/linux-automation/meta-lxatac) depends on the plain `python3-pyserial.bb` but that does not matter to much anyways, as I think the issue lies with `python3-pyserial-labgrid.bb`.

What would be a good fix for this? Installing the forked `pyserial` as `pyserial-labgrid` and importing it accordingly in labgrid?

```
Installing python3-pyserial (3.5) on root
Downloading file:meta-lxatac/build/tmp/work/lxatac-oe-linux-gnueabi/lxatac-core-image-base/1.0-r0/oe-rootfs-repo/cortexa7t2hf-neon-vfpv4/python3-pyserial_3.5-r0_cortexa7t2hf-neon-vfpv4.ipk.
Installing python3-charset-normalizer (3.0.0) on root
Downloading file:meta-lxatac/build/tmp/work/lxat * check_data_file_clashes: Package python3-pyserial-labgrid wants to install file meta-lxatac/build/tmp/work/lxatac-oe-linux-gnueabi/lxatac-core-image-base/1.0-r0/rootfs/usr/lib/python3.10/site-packages/serial/__init__.py
        But that file is already provided by package  * python3-pyserial
 * check_data_file_clashes: Package python3-pyserial-labgrid wants to install file meta-lxatac/build/tmp/work/lxatac-oe-linux-gnueabi/lxatac-core-image-base/1.0-r0/rootfs/usr/lib/python3.10/site-packages/serial/__main__.py
…
```

Bastian-Krause commented:
I guess this is similar to https://github.com/labgrid-project/labgrid/issues/1245.

Bastian-Krause commented:
Let's postpone the version bump of python3-pyserial-labgrid. I've dropped the corresponding commit, rebased everything on latest master and bumped sispmctl to v4.11 instead of v4.10.

@hnez Could you please test this again?

Bastian-Krause commented:
Note: build failure will be fixed by #41.

ejoerns commented:
I've rebased this onto master with merged #41 to let the CI test it.

ejoerns commented on recipes-devtools/python/python3-janus_1.0.0.bb:
```diff
@@ -3,8 +3,8 @@ HOMEPAGE = "https://pypi.org/project/janus/"
+SRC_URI[sha256sum] = "df976f2cdcfb034b147a2d51edfc34ff6bfb12d4e2643d3ad0e10de058cb1612"
 
-inherit pypi setuptools3
+inherit python_setuptools_build_meta pypi
```
Maybe give a short hint about the adapted `inherit` in the commit message

Bastian-Krause replied:
Done.

hnez commented:
Hi,

> Could you please test this again?

the LXA TAC images (based on my mickledore development branch) build fine now and the `labgrid-exporter` seems to be working.
Running `usbmuxctl -h` also works, but I do not have an USB-Mux at hand to check if it works, but I would assume it does.

ejoerns reviewed (approved):
Looks fine for me from the Yocto integration perspective.

hnez commented:
I think this should be merged, shouldn't it?
